src/spikeinterface/core/channelsaggregationrecording.py

```diff
@@ -49,14 +62,16 @@ def __init__(self, recording_list, renamed_channel_ids=None):
                 break
 
         if not (ok1 and ok2 and ok3 and ok4):
-            raise ValueError("Sortings don't have the same sampling_frequency/num_segments/dtype/num samples")
+            raise ValueError(
+                "Recordings do not have consistent sampling frequency, number of segments, data type, or number of samples."
```

Comment:
I understand the check so feel free to ignore this comment, but the error doesn't really help the user as is. It says "something is wrong", but the user doesn't know. Maybe I have a recording I converted to a different dtype and so I assume it is okay, but don't realize. To make better errors would be to be clear what is wrong, but then that makes the check into a series of if

if not ok1:
raise error for sampling
if not ok2:
raise error for num seg.

But this isn't in the scope of this particular PR, so feel free to ignore.

Reply:
I wanted to modify this one but Sam might like it that way and I did not wanted to increase the scope of the discussion. In my opinion it would be better to have an assertion per check.

Reply:
Sounds good. Mental note for a future PR :)